rs/state_manager/src/lib.rs

```diff
@@ -1139,6 +1171,13 @@ impl PageMapType {
             PageMapType::WasmChunkStore(id) => state
                 .canister_state_mut(id)
                 .map(|can| can.system_state.wasm_chunk_store.page_map_mut()),
+            // Snapshots are kept behind an `Arc` inside `state`, which makes the pattern of `list_all` and `get_mut` for pagemaps difficult
+            // without excessive copies.
+            // In the few places where we need to iterate over all pagemaps and modify them we instead handle snapshots separately.
+            // Note that this is not a problem in the many more places where we need to iterate over all pagemaps and only need read access.
+            PageMapType::SnapshotWasmMemory(_) => unimplemented!(),
+            PageMapType::SnapshotStableMemory(_) => unimplemented!(),
+            PageMapType::SnapshotWasmChunkStore(_) => unimplemented!(),
```

Comment:
I wonder whether splitting the PageMapType into PageMapType aas here plus SnapshotPageMApType would make it easier, as snapshots need a different treatment and can be a different type to make it explicit. But not sure that could work.

Reply:
Yeah I am not particularly happy with this bit either. I was trying out a few things but didn't really find anything that worked.

I am very open to revisiting this part of the code again, but if it's ok I wouldn't do that as part of this PR.